tasks.c

```diff
@@ -7678,3 +7678,44 @@ static void prvAddCurrentTaskToDelayedList( TickType_t xTicksToWait,
     #endif
 
 #endif /* if ( configINCLUDE_FREERTOS_TASK_C_ADDITIONS_H == 1 ) */
+/*-----------------------------------------------------------*/
+
+/*
+  vApplicationGetIdleTaskMemory gets called when configSUPPORT_STATIC_ALLOCATION
+  equals to 1 and is required for static memory allocation support.
+  This definition can be overwritten. 
+*/
+
+#if (configSUPPORT_STATIC_ALLOCATION == 1)
+
+__attribute__((__weak__)) void vApplicationGetIdleTaskMemory (StaticTask_t **ppxIdleTaskTCBBuffer,
```

Comment:
It looks good to me, except that Visual C doesn't recognise `__attribute__((__weak__))`.
Visual C is used to compile and run demos for both FreeRTOS and for +TCP.
You might get it working when using `#pragma WEAK` in case

Reply:
Thanks for your suggestion, but as FreeRTOS is used with many different compilers, compiler specific syntax can only appear in the port layer (i.e. in the C files that are only built with which ever compiler's syntax is used).

Reply:
Thanks for the reviews. So should it be added to GCC port directory?
There exists a `/portable/ThirdParty/GCC/RP2040/idle_task_static_memory.c`, but it doesn't have weak attribute and also shouldn't it be made visible using `extern` ?

Reply:
I do not think that would be a good place either because we will need to have the same code for all the supported compilers. I think it is best handled in the documentation, which we have done here - https://www.freertos.org/a00110.html#configSUPPORT_STATIC_ALLOCATION. 

> There exists a /portable/ThirdParty/GCC/RP2040/idle_task_static_memory.c

That is a third party port and not from us.

> shouldn't it be made visible using extern ?

Unless explicitly marked `static`, a function is by default visible.